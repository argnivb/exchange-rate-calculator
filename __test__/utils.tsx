import { render } from '@testing-library/react';
import Providers from '@/providers';

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { customRender as render };
