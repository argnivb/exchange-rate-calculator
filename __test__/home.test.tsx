import Home from '@/app/page';
import { render, screen } from './utils';

describe('Home', () => {
  it('renders a heading', () => {
    render(<Home />);

    const heading = screen.getByRole('heading', {
      name: /Exchange Rate Calculator/i,
    });

    expect(heading).toBeInTheDocument();
  });

  it('renders ExchangeRateCalculator', () => {
    render(<Home />);

    const exchangeRateCalculatorElement = screen.getByTestId(
      'exchange-rate-calculator'
    );
    expect(exchangeRateCalculatorElement).toBeInTheDocument();
  });

  it('renders Home page unchanged', async () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
