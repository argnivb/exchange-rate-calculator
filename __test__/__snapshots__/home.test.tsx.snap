// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders Home page unchanged 1`] = `
.c2 {
  margin-left: auto;
  margin-right: auto;
  max-width: 374px;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 30px;
  padding: 32px;
  padding-bottom: 64px;
  margin-bottom: 128px;
  background-color: #ffffff;
  box-shadow: 0 1px 12px #D8EBFF;
}

.c4 {
  font-family: fontFamily;
  font-weight: 700;
  font-size: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 32px;
}

.c5 {
  font-family: fontFamily;
  font-weight: 700;
  font-size: 24px;
  background-color: #fff;
  outline: none;
  border: 1px solid #e5e5e5;
  border-radius: 30px;
  padding: 16px 32px;
  width: 100%;
}

.c5:focus-visible {
  border-color: #1E90FF;
}

.c6 {
  position: absolute;
  bottom: -1.8rem;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  background-color: #1E90FF;
  color: #fff;
  -webkit-transition: background-color 0.3s;
  transition: background-color 0.3s;
  border: 1px solid rgba(0,0,0,0);
  border-radius: 30px;
  outline: none;
  font-family: fontFamily;
  font-weight: 700;
  font-size: 24px;
  padding: 16px 32px;
}

.c6:hover {
  background-color: #0066CC;
}

.c6:focus-visible {
  outline: 1px solid #1E90FF;
  outline-offset: 0.125rem;
}

.c6:active {
  background-color: #1E90FF;
}

.c6:disabled {
  background-color: #aaaaaa;
}

.c6:disabled,
.c6:disabled:hover {
  cursor: default;
}

.c7 {
  font-family: fontFamily;
  font-weight: 700;
  font-size: 48px;
  text-align: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 64px;
  padding-top: 64px;
}

.c1 {
  font-family: fontFamily;
  font-size: 36px;
  font-weight: 700;
  text-align: center;
}

<div>
  <main
    class="c0"
  >
    <h1
      class="c1"
    >
      Exchange Rate Calculator
    </h1>
    <div
      class="c2"
      data-testid="exchange-rate-calculator"
    >
      <form>
        <div
          class="c3"
        >
          <div
            class="c4"
          >
            <p>
              USD
            </p>
            <span>
              to
            </span>
            <p>
              EUR
            </p>
          </div>
          <input
            class="c5"
            name="quantity"
            placeholder="Amount"
            type="text"
          />
          <button
            class="c6"
            type="submit"
          >
            Convert
          </button>
        </div>
      </form>
      <p
        class="c7"
      >
        18825.10
      </p>
    </div>
  </main>
</div>
`;
